src/adagrams.js

```diff
@@ -1,15 +1,101 @@
+const letterDistribution = {
+  A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2, I: 9, J: 1, K: 1, L: 4, M: 2,
+  N: 6, O: 8, P: 2, Q: 1, R: 6, S: 4, T: 6, U: 4, V: 2, W: 2, X: 1, Y: 2, Z: 1
+};
+
+
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const letterPool = Object.keys(letterDistribution);
+  const hand = [];
+
+
+  for (let count = 0; count < 10; count++) {
+    const randomIndex = Math.floor(Math.random() * letterPool.length);
+    const letter = letterPool[randomIndex];
+
+    if (letterDistribution[letter] > 0) {
+      hand.push(letter);
+      letterDistribution[letter] -= 1; 
+      
+    } 
+    if (letterDistribution[letter] === 0) {
+      letterPool.splice(randomIndex, 1);
+    }
```

Comment:
Modifying the `letterDistribution` dictionary is causing issues. Since the dictionary is defined outside of the function, it is never re-built. So each time you run the `drawLetters()` function, letters are permanently removed from the list until eventually there are no letters left. To fix this, either make a copy of the dictionary at the beginning of the function or move the `letterDistribution` dictionary into the function.